lib/pester.rb

```diff
@@ -56,12 +55,20 @@ def self.retry_action(opts = {}, &block)
         result = yield block
         return result
       rescue => e
-        class_reraise = opts[:retry_error_classes] && !opts[:retry_error_classes].include?(e.class)
+        retry_error_classes = opts[:retry_error_classes]
+        retry_error_messages = opts[:retry_error_messages]
+
+        class_retry = retry_error_classes && retry_error_classes.include?(e.class)
+        if class_retry && retry_error_messages && !retry_error_messages.any? { |m| e.message[m] }
+          class_retry = false
+        end
```

Comment:
that doesn't quite cut it unfortunately but yeah i agree it feels too
confusing.

if there ARE NOT provided retry_error_classes, a retry_error_messages entry
would be sufficient to cause a retry on its own.

if there ARE provided retry_error_classes AND retry_error_messages, both
the class and the message should match to retry (this is how a lot of our
retries are in the DW/now phorklift)

On Wed, Apr 1, 2015 at 12:20 PM, dollschasingmen <notifications@github.com>
wrote:

> In lib/pester.rb
> <https://github.com/lumoslabs/pester/pull/5#discussion_r27602011>:
>
> > @@ -56,12 +55,20 @@ def self.retry_action(opts = {}, &block)
> >          result = yield block
> >          return result
> >        rescue => e
> > -        class_reraise = opts[:retry_error_classes] && !opts[:retry_error_classes].include?(e.class)
> > +        retry_error_classes = opts[:retry_error_classes]
> > +        retry_error_messages = opts[:retry_error_messages]
> > +
> > +        class_retry = retry_error_classes && retry_error_classes.include?(e.class)
> > +        if class_retry && retry_error_messages && !retry_error_messages.any? { |m| e.message[m] }
> > +          class_retry = false
> > +        end
>
> i find this confusing -- maybe this more straightfoward?
> class_retry = retry_error_classes && retry_error_classes.include?(e.class)
> message_retry = retry_error_messages && !retry_error_messages.any? { |m|
> e.message[m] }
> retry = message_retry && class_retry
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lumoslabs/pester/pull/5/files#r27602011>.
>